src/dbnode/client/write_state.go

```diff
@@ -213,3 +249,30 @@ func (p *writeStatePool) Get() *writeState {
 func (p *writeStatePool) Put(w *writeState) {
 	p.pool.Put(w)
 }
+
+func getShardState(shardState shard.State, leavingCountsIndividually bool, leavingAndInitializingCountsAsPair bool) shardStatus {
```

Comment:
two nits:
(1) don't use `get...` as a prefix in go methods - go's idioms are to prefer to not that for getters, and when you're creating new states, call it `new...` instead
(2) break this decl into a line per argument to make it easier to read